activerecord/test/cases/locking_test.rb

```diff
@@ -738,6 +738,35 @@ def test_with_lock_rolls_back_transaction
       assert_equal old, person.reload.first_name
     end
 
+    def test_with_lock_configures_transaction
+      person = Person.find 1
+      Person.transaction do
+        outer_transaction = Person.connection.transaction_manager.current_transaction
+        assert_equal true, outer_transaction.joinable?
+        person.with_lock(requires_new: true, joinable: false) do
```

Comment:
The updated test no longer tests `isolation:`. Testing isolation is database-specific because they each support different isolation levels (and sqlite3 doesn't even like to have isolation specified at all since it only supports one isolation level). We forward all opts to `transaction` anyway, so if any work, all should work. I experimented with a test that `assert_raises ActiveRecord::TransactionIsolationError` when provided a nonexistent isolation type, which would work consistently across DB engines, but it felt weird to test that behavior from here.